File: Data Structures & Algorithms/two-integer-sum/submission-4.py
class Solution:
    # 3, 4, 5, 6 - 7
    # 
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        storage = {}

        for i, n in enumerate(nums):
            diff = target - n
            if diff in storage:
                return [storage[diff], i]
            storage[n] = i

        return []

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-4.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int minimum = Integer.MAX_VALUE;

        int sum = 0;
        int left = 0;

        for (int i = 0; i < nums.length; i++) {
            final int n = nums[i];
            sum += n;

            while (left < nums.length && sum >= target) {
                minimum = Math.min(minimum, (i - left) + 1);
                sum -= nums[left];
                left++;
            }
        }

        return minimum == Integer.MAX_VALUE ? 0 : minimum;
    }
}